core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/aird/AssociationCDBMigrationContributor.java

```diff
@@ -0,0 +1,376 @@
+        containerVariable = dEdgeContainer.getTarget();
+      } else {
+        containerVariable = null;
+      }
+      IInterpreter interpreter = session.getInterpreter();
+      interpreter.setVariable(IInterpreterSiriusVariables.DIAGRAM, dEdge.getParentDiagram());
+      interpreter.setVariable(IInterpreterSiriusVariables.VIEW, dEdge);
+      interpreter.setVariable(IInterpreterSiriusVariables.SOURCE_VIEW, dEdge.getSourceNode());
+      interpreter.setVariable(IInterpreterSiriusVariables.TARGET_VIEW, dEdge.getTargetNode());
+
+      StyleDescription styleDescription = new MappingWithInterpreterHelper(interpreter).getBestStyleDescription(
+          dEdge.getDiagramElementMapping(), dEdge.getTarget(), dEdge, containerVariable, dEdge.getParentDiagram());
+
+      interpreter.unSetVariable(IInterpreterSiriusVariables.DIAGRAM);
+      interpreter.unSetVariable(IInterpreterSiriusVariables.VIEW);
+      interpreter.unSetVariable(IInterpreterSiriusVariables.SOURCE_VIEW);
+      interpreter.unSetVariable(IInterpreterSiriusVariables.TARGET_VIEW);
+
+      dEdge.getOwnedStyle().setDescription(styleDescription);
+    });
+  }
+
+  private void migrateDEdge(DEdge dEdge) {
+    EdgeTarget sourceNode = dEdge.getSourceNode();
+    EdgeTarget targetNode = dEdge.getTargetNode();
+    String beginLabel = dEdge.getBeginLabel();
+    String endLabel = dEdge.getEndLabel();
+    EdgeArrows sourceArrow = dEdge.getOwnedStyle().getSourceArrow();
+    EdgeArrows targetArrow = dEdge.getOwnedStyle().getTargetArrow();
+    List<String> customFeatures = dEdge.getOwnedStyle().getCustomFeatures();
+
+    List<DEdge> sourceNodeIncommingEdges = sourceNode.getIncomingEdges();
+    List<DEdge> sourceNodeOutgoingEdges = sourceNode.getOutgoingEdges();
+    List<DEdge> targetNodeIncommingEdges = targetNode.getIncomingEdges();
+    List<DEdge> targetNodeOutgoingEdges = targetNode.getOutgoingEdges();
+
+    dEdge.setSourceNode(targetNode);
+    dEdge.setTargetNode(sourceNode);
+    dEdge.setBeginLabel(endLabel);
+    dEdge.setEndLabel(beginLabel);
+    sourceNodeIncommingEdges.add(dEdge);
+    targetNodeOutgoingEdges.add(dEdge);
+    sourceNodeOutgoingEdges.remove(dEdge);
+    targetNodeIncommingEdges.remove(dEdge);
+
+    dEdge.getOwnedStyle().setSourceArrow(targetArrow);
+    dEdge.getOwnedStyle().setTargetArrow(sourceArrow);
+
+    // inverse the custom features sourceArrow and targetArrow
+    String sourceArrowFeatureName = DiagramPackage.eINSTANCE.getEdgeStyle_SourceArrow().getName();
+    String targetArrowFeatureName = DiagramPackage.eINSTANCE.getEdgeStyle_TargetArrow().getName();
+    if (customFeatures.contains(sourceArrowFeatureName) && !customFeatures.contains(targetArrowFeatureName)) {
+      customFeatures.add(targetArrowFeatureName);
+      customFeatures.remove(sourceArrowFeatureName);
+    } else if (!customFeatures.contains(sourceArrowFeatureName) && customFeatures.contains(targetArrowFeatureName)) {
```

Comment:
If we go there, we are going to compute again the contains for both. computing it once and using local variables would gain a little computation time

Reply:
done